app/models/fasta_record.rb

```diff
@@ -38,11 +38,17 @@ def self.build_fasta_record(line)
 
     ActiveRecord::Base.logger.info("New fasta record: #{strain}")
 
-    dg = get_dg(country, division, location, region)
-
-    FastaRecord.new(strain: strain, gisaid_epi_isl: gisaid_epi_isl,
-                    genbank_accession: genbank_accession, detailed_geo_location: dg,
-                    date_collected: date, variant_name: variant_name)
+    dg, dg_id = get_dg(country, division, location, region)
```

Comment:
not sure i understand what's going on here... dg is saved now  so it should be the same in both
is the whole fix really in the cache key?